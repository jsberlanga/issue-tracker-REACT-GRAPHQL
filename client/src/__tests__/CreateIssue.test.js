import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CreateIssueComponent from "../components/CreateIssue";
import { MockedProvider } from "react-apollo/test-utils";
import { BrowserRouter } from "react-router-dom";

const fakeIssue = {
  id: "123qwe",
  title: "Title for a Fake Issue",
  description: "Description for a Fake Issue",
  status: "OPEN"
};

describe("<CreateIssue />", () => {
  it("renders and matches snapshot", async () => {
    const wrapper = mount(
      <MockedProvider>
        <BrowserRouter>
          <CreateIssueComponent />
        </BrowserRouter>
      </MockedProvider>
    );
    const form = wrapper.find(".form");
    expect(toJSON(form)).toMatchSnapshot();
    // console.log(wrapper.debug());
  });
  it("handles state updates", async () => {
    const wrapper = mount(
      <MockedProvider>
        <BrowserRouter>
          <CreateIssueComponent />
        </BrowserRouter>
      </MockedProvider>
    );
    wrapper.find("#title").simulate("change", {
      target: { value: "Title Testing", name: "title" }
    });

    wrapper.find("#description").simulate("change", {
      target: { value: "Description Testing", name: "description" }
    });

    // Error returns ReactWrapper empty {} - Cannot read property 'formData' of null
    // expect(wrapper.find("CreateIssue").instance().formData..toMatchObject({...})
  });
});
