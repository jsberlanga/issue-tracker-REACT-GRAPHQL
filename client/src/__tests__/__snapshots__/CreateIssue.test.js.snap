// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateIssue /> renders and matches snapshot 1`] = `
<form
  className="form form__create"
  onSubmit={[Function]}
>
  <label
    htmlFor="Title"
  >
    Title
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Enter a title for the issue"
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="Description"
  >
    Description
    <textarea
      id="description"
      name="description"
      onChange={[Function]}
      placeholder="Enter a description"
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="Status"
  >
    Status
    <select
      name="status"
      onChange={[Function]}
    >
      <option
        key="Select the current status"
        value="Select the current status"
      >
        Select the current status
      </option>
      <option
        key="OPEN"
        value="OPEN"
      >
        OPEN
      </option>
      <option
        key="PENDING"
        value="PENDING"
      >
        PENDING
      </option>
      <option
        key="COMPLETED"
        value="COMPLETED"
      >
        COMPLETED
      </option>
    </select>
  </label>
  <button
    className="btn__primary"
  >
    Create
  </button>
</form>
`;
